src/Microsoft.AspNet.Mvc.TagHelpers/Logging/ModeMatchResultLoggerExtensions.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.Mvc.TagHelpers.Internal;
+using Microsoft.AspNet.Razor.TagHelpers;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers.Logging
+{
+    public static class ModeMatchResultLoggerExtensions
+    {
+        /// <summary>
+        /// Log values for <see cref="AspNet.Razor.TagHelpers.ITagHelper"/> instances that opt out of
+        /// processing due to missing attributes for one of several possible modes.
+        /// </summary>
+        private class PartialModeMatchLogValues<TMode> : ILogValues
```

Comment:
Got super confused by this file since the `private class` is at the top. We typically put private classes at the end of our public classes.